pytorch_lightning/trainer/trainer.py

```diff
@@ -1095,7 +1095,8 @@ def _pre_training_routine(self):
         # register auto-resubmit when on SLURM
         self.slurm_connector.register_slurm_signal_handlers()
 
-        self.checkpoint_connector.resume_end()
+        if self.state.fn == TrainerFn.FITTING:
+            self.checkpoint_connector.resume_end()
```

Comment:
why do we need this check on fitting for resume end?

Reply:
because `trainer.tune` calls it too. so just don't want it to delete the reloaded_state_dict in case of tuning. Also, `resume_start` is under the same condition as well.

Reply:
should we explicitly call this with tune in mind then? 
```suggestion
        if self.state.fn != TrainerFn.TUNING:
            self.checkpoint_connector.resume_end()
```

Reply:
will update.